zebrad/src/commands/seed.rs

```diff
@@ -39,37 +37,27 @@ impl Service<Request> for SeedService {
     fn poll_ready(&mut self, _cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
         debug!("SeedService.state: {:?}", self.state);
 
-        let mut poll_result = Poll::Pending;
-
-        // We want to be able to consume the state, but it's behind a mutable
-        // reference, so we can't move it out of self without swapping in a
-        // placeholder, even if we immediately overwrite the placeholder.
-        let tmp_state = std::mem::replace(&mut self.state, SeederState::TempState);
-
-        self.state = match tmp_state {
-            SeederState::AwaitingAddressBook(mut rx) => match rx.try_recv() {
+        match self.state {
+            SeederState::Ready(_) => return Poll::Ready(Ok(())),
+            SeederState::AwaitingAddressBook(ref mut rx) => match rx.try_recv() {
+                Err(e) => {
+                    error!("SeedService oneshot sender dropped: {:?}", e);
+                    return Poll::Ready(Err(e.into()));
+                }
+                Ok(None) => {
+                    debug!("SeedService got a message with `None` in it. 🤔");
```

Comment:
I think that `Ok(None)` means "no message yet" rather than "got `None` message"

Reply:
Ah, gotcha.
```suggestion
                    trace!("SeedService hasn't received a message via the oneshot yet.");
```